fix: updating re-added queries works (+test)

svenrdz opened the pull request:
Resolves https://github.com/ESGF/esgf-download/issues/96

Fixes a bug where linking would not happen for files that were already in the database, even if they were not linked to the query that is updated.

Simplified the logic for `update`:
* always add new files/datasets to database
* then prompt for whether to both link to query and send to download queue
* linking is always done on all files, regardless of whether they were already in db

AtefBN reviewed (approved):
Mon seul commentaire c'est que je pense registered/unregistered file/dataset ou bien recorded/unrecorded c'est peut être plus explicite que known/unknown pour quelqu'un qui touche au code en abstrait mais c'est rien en vrai

svenrdz commented:
@AtefBN 

> Mon seul commentaire c'est que je pense registered/unregistered file/dataset ou bien recorded/unrecorded c'est peut être plus explicite que known/unknown pour quelqu'un qui touche au code en abstrait mais c'est rien en vrai

Updated, and I also fixed a regression on update speed and a failing test regarding query timestamps
